common/src/main/java/org/vivecraft/api/client/VivecraftClientAPI.java

```diff
@@ -0,0 +1,217 @@
+package org.vivecraft.api.client;
+
+import com.google.common.annotations.Beta;
+import net.minecraft.world.InteractionHand;
+import org.vivecraft.api.client.data.VRPoseHistory;
+import org.vivecraft.api.data.VRData;
+import org.vivecraft.client.api_impl.ClientAPIImpl;
+
+import javax.annotation.Nullable;
+
+public interface VivecraftClientAPI {
+
+    static VivecraftClientAPI getInstance() {
+        return ClientAPIImpl.INSTANCE;
+    }
+
+    /**
+     * Adds the tracker to the list of all trackers to be run for the local player. See the documentation for
+     * {@link Tracker} for more information on what a tracker is.
+     *
+     * @param tracker Tracker to register.
+     */
+    void addTracker(Tracker tracker);
+
+    /**
+     * Gets data representing the devices as they exist in the room before the game tick.
+     * Note that this data is gathered BEFORE mod loaders' pre-tick events.
+     *
+     * @return Data representing the devices in the room pre-tick.
+     * @throws IllegalStateException Thrown when the local player isn't in VR.
+     */
+    VRData getPreTickRoomData() throws IllegalStateException;
```

Comment:
Should we return null when the player isn't in VR instead of throwing an exception? Thinking what would be the best for users of the api

Reply:
That's something I'm not sure of myself, and I think goes well either-way (see `VivecraftAPI`, where `null` is returned instead of throwing an exception). I think for consistency with the common API, it'd make sense for me to change this to return `null` instead of asking API consumers to be ready to handle an exception.

Reply:
Resolved in https://github.com/hammy275/VivecraftMod/commit/0f667b0f452be1b68a4e418d1b726d490121c1a7 . Let me know if you think we should walk back to using `IllegalStateException`s!